Mainnet/STOContract/STOContract/STOContract.cs

```diff
@@ -0,0 +1,1396 @@
+    /// </summary>
+    /// <param name="tokenId">ID of the NFT to validate.</param>
+    private void CanOperate(ulong tokenId)
+    {
+        Address tokenOwner = GetIdToOwner(tokenId);
+        Assert(tokenOwner == this.Message.Sender || GetOwnerToOperator(tokenOwner, this.Message.Sender));
+    }
+
+    /// <summary>
+    /// Guarantees that the msg.sender is allowed to transfer NFT.
+    /// </summary>
+    /// <param name="tokenId">ID of the NFT to transfer.</param>
+    private void CanTransfer(ulong tokenId)
+    {
+        Address tokenOwner = GetIdToOwner(tokenId);
+        Assert(
+          tokenOwner == this.Message.Sender
+          || GetIdToApproval(tokenId) == Message.Sender
+          || GetOwnerToOperator(tokenOwner, Message.Sender)
+        );
+    }
+
+    /// <summary>
+    /// Guarantees that tokenId is a valid Token.
+    /// </summary>
+    /// <param name="tokenId">ID of the NFT to validate.</param>
+    private void ValidNFToken(ulong tokenId)
+    {
+        Address tokenOwner = GetIdToOwner(tokenId);
+        EnsureAddressIsNotEmpty(tokenOwner);
+    }
+
+    /// <summary>
+    /// Sets the contract owner who can mint/bur
+    /// </summary>
+    /// <param name="owner"></param>
+    public void TransferOwnership(Address owner)
+    {
+        EnsureOwnerOnly();
+        Assert(owner != Address.Zero, $"The {nameof(owner)} parameter can not be default(zero) address.");
+
+        Log(new OwnershipTransferedLog { PreviousOwner = this.Owner, NewOwner = owner });
+
+        this.Owner = owner;
+    }
+
+    private void EnsureOwnerOnly()
+    {
+        Assert(Message.Sender == Owner, "Only owner of the contract can set new owner.");
+    }
+
+    /// <summary>
+    /// Mints new tokens
+    /// </summary>
+    /// <param name="address">The address that will own the minted NFT</param>
+    /// <param name="amount">Number of tokens will be created</param>
+    public void MintAll(Address address, ulong amount)
+    {
+        EnsureOwnerOnly();
+        EnsureAddressIsNotEmpty(address);
+        Assert(amount > 0, "the amount should be higher than zero");
+
+        var index = TotalSupply;
+        var lastIndex = checked(index + amount);
+        var tokenId = NextTokenId;
+
+        while (index < lastIndex)
+        {
+            AddNFToken(address, tokenId);
+            SetTokenByIndex(index, tokenId);
+            SetIndexByToken(tokenId, index);
+
+            Log(new TransferLog { From = Address.Zero, To = address, TokenId = tokenId });
+
+            checked
+            {
+                index++;
+                tokenId++;
+            }
+        }
+
+        TotalSupply = checked(TotalSupply + amount);
+        NextTokenId = tokenId;
+    }
+
+    public void Burn(ulong tokenId)
+    {
+        Address tokenOwner = GetIdToOwner(tokenId);
+
+        EnsureAddressIsNotEmpty(tokenOwner);
+
+        Assert(tokenOwner == Message.Sender, "Only token owner can burn the token.");
```

Comment:
I think you fixed this in the other PR, no need to check for AddressIsNotEmpty here because token == Message.Sender is good enough.

Probably worth copying over all 3 token types back into this PR just to make sure all changes are up to date here.